test/hatvaults.js

```diff
@@ -1184,13 +1186,55 @@ contract("HatVaults", (accounts) => {
       assertVMException(ex, "BountyPercentageHigherThanMaxBounty");
     }
     var tx = await hatVaults.claimReward(0, { from: staker });
-    assert.equal(tx.logs[0].event, "ClaimReward");
-    assert.equal(tx.logs[0].args._pid, 0);
+    assert.equal(tx.logs[1].event, "ClaimReward");
+    assert.equal(tx.logs[1].args._pid, 0);
+
+    assert.equal(
+      (await hatToken.balanceOf(hatVaults.address)).toString(),
+      new web3.utils.BN(web3.utils.toWei(hatVaultsExpectedHatsBalance.toString())).sub(expectedReward).toString()
+    );
+    assert.equal(
+      (await hatToken.balanceOf(staker)).toString(),
+      expectedReward.toString()
+    );
+    assert.equal(await stakingToken.balanceOf(staker), 0);
+    assert.equal(
+      await stakingToken.balanceOf(hatVaults.address),
+      web3.utils.toWei("1")
+    );
+  });
 
+  it("cannot claim reward twice", async () => {
+    await setup(
+      accounts,
+      (await web3.eth.getBlock("latest")).number,
+      8000,
+      [6000, 2000, 500, 0, 1000, 500],
+      10000
+    );
+    var staker = accounts[1];
+    await stakingToken.approve(hatVaults.address, web3.utils.toWei("4"), {
+      from: staker,
+    });
+    await stakingToken.mint(staker, web3.utils.toWei("1"));
+    await hatVaults.deposit(0, web3.utils.toWei("1"), { from: staker });
     assert.equal(
       await hatToken.balanceOf(hatVaults.address),
       web3.utils.toWei(hatVaultsExpectedHatsBalance.toString())
     );
+
+    assert.equal(await hatToken.balanceOf(staker), 0);
+
+    let expectedReward = await calculateExpectedReward(staker, 1);
+    await hatVaults.claimReward(0, { from: staker });
+    var tx = await hatVaults.claimReward(0, { from: staker });
+    assert.equal(tx.logs[1].event, "ClaimReward");
+    assert.equal(tx.logs[1].args._pid, 0);
+
+    assert.equal(
```

Comment:
maybe better to check the balances after the first claim reward, and then check that it hasn't changed after the second one

Reply:
Well I did it like this because it actually does change, by one block of reward

Reply:
Ok I made it to claim twice in the same block